code/modules/client/preference_setup/origin/origin.dm

```diff
@@ -0,0 +1,249 @@
+
+/datum/category_item/player_setup_item/origin
+	name = "Origin"
+	sort_order = 1
+
+/datum/category_item/player_setup_item/origin/load_character(var/savefile/S)
+	S["culture"]    		>> pref.culture
+	S["origin"]     		>> pref.origin
+	S["citizenship"]		>> pref.citizenship
+	S["religion"]			>> pref.religion
+	S["accent"]				>> pref.accent
+	S["economic_status"] 	>> pref.economic_status
+
+/datum/category_item/player_setup_item/origin/save_character(var/savefile/S)
+	S["culture"]			<< pref.culture
+	S["origin"]				<< pref.origin
+	S["citizenship"]		<< pref.citizenship
+	S["religion"]			<< pref.religion
+	S["accent"]				<< pref.accent
+	S["economic_status"]	<< pref.economic_status
+
+/datum/category_item/player_setup_item/origin/gather_load_query()
+	return list(
+		"ss13_characters" = list(
+			"vars" = list(
+				"culture",
+				"origin",
+				"economic_status",
+				"citizenship",
+				"religion",
+				"accent"
+			),
+			"args" = list("id")
+		)
+	)
+
+/datum/category_item/player_setup_item/origin/gather_load_parameters()
+	return list("id" = pref.current_character)
+
+/datum/category_item/player_setup_item/origin/gather_save_query()
+	return list(
+		"ss13_characters" = list(
+			"culture",
+			"origin",
+			"economic_status",
+			"citizenship",
+			"religion",
+			"accent",
+			"id" = 1,
+			"ckey" = 1
+		)
+	)
+
+/datum/category_item/player_setup_item/origin/gather_save_parameters()
+	return list(
+		"culture" = pref.culture,
+		"origin" = pref.origin,
+		"economic_status" = pref.economic_status,
+		"citizenship" = pref.citizenship,
+		"religion" = pref.religion,
+		"accent" = pref.accent,
+		"id" = pref.current_character,
+		"ckey" = PREF_CLIENT_CKEY
+	)
+
+//Maybe turn accents/religion/citizenship into decls?
+//Todo: remove species accent/citizenship lists.
+//Maybe todo: add a proc that returns the default/most common accent/religion/citizenship for a certain origin so we don't have to do list[1].
+/datum/category_item/player_setup_item/origin/sanitize_character(var/sql_load = 0)
+	var/datum/species/S = all_species[pref.species]
+	var/decl/origin_item/CI = S.get_default_culture()
+	if(!istext(pref.culture))
+		pref.culture = "[CI]"
+	var/decl/origin_item/culture/our_culture = decls_repository.get_decl(text2path(pref.culture))
+	if(!istext(pref.origin))
+		var/decl/origin_item/OI = pick(our_culture.possible_origins)
+		pref.origin = "[OI]"
+	else
+		var/decl/origin_item/origin/origin_check = text2path(pref.origin)
+		if(!(origin_check in our_culture.possible_origins))
+			to_client_chat(SPAN_WARNING("Your origin has been reset due to it being incompatible with your culture!"))
+			var/decl/origin_item/OI = pick(our_culture.possible_origins)
+			pref.origin = "[OI]"
+	var/decl/origin_item/origin/our_origin = decls_repository.get_decl(text2path(pref.origin))
+	if(!pref.citizenship)
+		pref.citizenship = our_origin.possible_citizenships[1]
+	if(!pref.religion)
+		pref.religion = our_origin.possible_religions[1]
```

Comment:
```suggestion

```

If these are null then `!(pref.citizenship in our_origin.possible_citizenships)` is `TRUE` so we don't need to do it here as well?